src/shell/commands/detect_hotkey.cpp

```diff
@@ -0,0 +1,116 @@
+    if (hotkey_type_check == "read") {
+        req.type = dsn::apps::hotkey_type::type::READ;
+    } else if (hotkey_type_check == "write") {
+        req.type = dsn::apps::hotkey_type::type::WRITE;
+    } else {
+        fmt::print(stderr, "\"{}\"  is a invalid hotkey type (read or write) \n", hotkey_type);
+        return false;
+    }
+    if (hotkey_action_check == "start") {
+        req.action = dsn::apps::hotkey_detect_action::START;
+    } else if (hotkey_action_check == "stop") {
+        req.action = dsn::apps::hotkey_detect_action::STOP;
+    } else {
+        fmt::print(
+            stderr, "\"{}\"  is a invalid hotkey detect action (start or stop) \n", hotkey_action);
+        return false;
+    }
+    return true;
+}
+
+// TODO: (Tangyanzhao) merge hotspot_partition_calculator::send_hotkey_detect_request
+bool detect_hotkey(command_executor *e, shell_context *sc, arguments args)
+{
+    // detect_hotkey [-a|--app_name] [-p|--partition_index][-c|--hotkey_action][-t|--hotkey_type]
+    const std::set<std::string> &params = {
+        "a", "app_name", "p", "partition_index", "c", "hotkey_action", "t", "hotkey_type"};
+    argh::parser cmd(args.argc, args.argv, argh::parser::PREFER_PARAM_FOR_UNREG_OPTION);
+    if (!validate_cmd(cmd, params)) {
+        return false;
+    }
+
+    std::string app_name = cmd({"-a", "--app_name"}).str();
+    uint64_t partition_index;
+    try {
+        partition_index = boost::lexical_cast<uint64_t>(cmd({"-p", "--partition_index"}).str());
+    } catch (boost::bad_lexical_cast &e) {
+        fmt::print(stderr,
+                   "\"{}\" is a invalid partition index \n",
+                   cmd({"-p", "--partition_index"}).str());
+        return false;
+    }
+    std::string hotkey_action = cmd({"-c", "--hotkey_action"}).str();
+    std::string hotkey_type = cmd({"-t", "--hotkey_type"}).str();
+
+    dsn::apps::hotkey_detect_request req;
+    if (!generate_hotkey_request(req, hotkey_action, hotkey_type)) {
+        return false;
+    }
+    auto request_ptr = std::make_unique<dsn::apps::hotkey_detect_request>(req);
+    auto resolver = partition_resolver::get_resolver(
+        sc->current_cluster_name.c_str(), sc->meta_list, app_name.c_str());
+    bool is_successful = true;
+    dsn::task_tracker tracker;
+    pegasus::server::detect_hotkey_rpc rpc(
+        std::move(request_ptr), RPC_DETECT_HOTKEY, std::chrono::seconds(10), partition_index);
+    rpc.call(resolver,
+             &tracker,
+             [app_name, partition_index, &is_successful](dsn::error_code error) {
+                 if (error != dsn::ERR_OK) {
+                     is_successful = false;
+                     fmt::print(stderr,
+                                "Hotkey detect rpc sending failed, in {}.{}, error_hint:{} \n",
+                                app_name,
+                                partition_index,
+                                error.to_string());
+                 }
+             })
+        ->wait();
```

Comment:
If partition_index is an invalid parameter, will it get_resolver succeed and send rpc succeed?
For example, this app partiion_count is 16, but partition_index is 18.